release_dashboard/lib/services/conductor.dart

```diff
@@ -14,15 +14,16 @@ abstract class ConductorService {
   /// Returns null when there is no active release, such as in the case when first initialized.
   pb.ConductorState? get state;
 
+  File get stateFile;
```

Comment:
Why was this added?

Reply:
Oh yeah, I should have added a comment about this. Because I am using the [`stateFile` getter](https://github.com/Yugue/cocoon/blob/5494852519fe5fc811d05665781f0c9f503d9d5b/release_dashboard/lib/services/local_conductor.dart#L31) of `local_conductor.dart` outside of the class in [here](https://github.com/Yugue/cocoon/blob/5494852519fe5fc811d05665781f0c9f503d9d5b/release_dashboard/lib/widgets/create_release_substeps.dart#L84). The getter wouldn't be accessible outside of its class without it being defined in the abstract class.

Reply:
Ahh, I see. I don't think that should be public or required as a dependency of calling `createRelease`. This is sort of a circular dependency situation, where you need to get a value from the ConductorService in order to give it back to the ConductorService. My recommendations:

1. For the `LocalConductorService` implementation, remove the `stateFile` parameter from `createRelease` (alternatively make it optional, and if it's not provided fallback to calling the getter)
2. Make the `stateFile` getter private, so that only the `LocalConductorService` itself can call it.

Reply:
Oh, thanks for pointing that out. That's a really good idea. Changed!!